src/commands/registries/azure/DockerWebhookCreateStep.ts

```diff
@@ -0,0 +1,83 @@
+import { RemoteTagTreeItem } from '../../../tree/registries/RemoteTagTreeItem';
+import { nonNullProp } from "../../../utils/nonNull";
+import { openExternal } from '../../../utils/openExternal';
+
+export class DockerWebhookCreateStep extends AzureWizardExecuteStep<IAppServiceWizardContext> {
+    public priority: number = 141; // execute after DockerSiteCreate
+    private _treeItem: RemoteTagTreeItem;
+    public constructor(treeItem: RemoteTagTreeItem) {
+        super();
+        this._treeItem = treeItem;
+    }
+
+    public async execute(context: IAppServiceWizardContext, progress: Progress<{
+        message?: string;
+        increment?: number;
+    }>): Promise<void> {
+        const site: Site = nonNullProp(context, 'site');
+        let siteClient = new SiteClient(site, context);
+        let appUri: string = (await siteClient.getWebAppPublishCredential()).scmUri;
+        if (this._treeItem.parent instanceof AzureRepositoryTreeItem) {
+            const creatingNewWebhook: string = `Creating webhook for web app "${context.newSiteName}"...`;
+            ext.outputChannel.appendLine(creatingNewWebhook);
+            progress.report({ message: creatingNewWebhook });
+            const webhook = await this.createWebhookForApp(this._treeItem, context, appUri);
+            ext.outputChannel.appendLine(`Created webhook "${webhook.name}" with scope "${webhook.scope}", id: "${webhook.id}" and location: "${webhook.location}"`);
+        } else if (this._treeItem.parent instanceof DockerHubRepositoryTreeItem) {
+            // point to dockerhub to create a webhook
+            // http://cloud.docker.com/repository/docker/<registryName>/<repoName>/webHooks
+            const dockerhubPrompt: string = "Copy & open";
+            const dockerhubUri: string = `https://cloud.docker.com/repository/docker/${this._treeItem.parent.parent.parent.username}/${this._treeItem.parent.repoName}/webHooks`;
+            let response: string = await vscode.window.showInformationMessage(`To set up a CI/CD webhook, open the page "${dockerhubUri}" and enter the URI to the created web app in your dockerhub account`, dockerhubPrompt);
+            if (response) {
+                vscode.env.clipboard.writeText(appUri);
+                // tslint:disable-next-line: no-floating-promises
+                openExternal(dockerhubUri);
+            }
+        }
+    }
+
+    public shouldExecute(context: IAppServiceWizardContext): boolean {
+        return !!context.site && (this._treeItem.parent instanceof AzureRepositoryTreeItem || this._treeItem.parent instanceof DockerHubRepositoryTreeItem);
+    }
+
+    private async createWebhookForApp(node: RemoteTagTreeItem, wizardContext: IActionContext & Partial<IAppServiceWizardContext>, appUri: string): Promise<Webhook | undefined> {
+        // fields derived from the app service wizard
+        let siteName: string = wizardContext.site.name;
```

Comment:
Rather than passing in wizardContext, I would just pass in the site in the execute method above.  If you are going to use wizardContext, don't you have nonNull check wizardContext.site like you did in line 33?

Reply:
Did you mean pass in just the site _from_ the execute method? I misread what you wrote initially.

Reply:
Yeah, instead of passing in the whole context, you could just pass in the site into createWebhookForApp.  You don't use the context for anything other than the site and you already did a nonNull check in the execute method.